scripts/notes.ts

```diff
@@ -0,0 +1,239 @@
+import { execSync } from 'node:child_process';
+import chalk from 'chalk';
+import readline from 'readline';
+import themes from '../themes.json';
+import { writeFileSync } from 'node:fs';
+
+interface PRDetails {
+  author: string;
+  ghpr: number;
+  hash: string;
+  labels: string[] | null;
+  ticket: string | null;
+  title: string;
+}
+
+type Themes = typeof themes | any;
+const theme: Themes = themes;
+
+const c = chalk;
+
+const icons = {
+  breaking: '🚨',
+  added: '🚀',
+  changed: '💅',
+  fixed: '🛠',
+  internal: '📦',
+};
+
+type Icons = typeof icons | any;
+const icon: Icons = icons;
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+/**
+ * Current milestone reference
+ */
+const cm = JSON.parse(execSync('gh api repos/CMSgov/design-system/milestones').toString())[0];
+
+/**
+ * Grabs latest tag from each theme and returns an array of `theme: version` items.
+ */
+export const getLatestVersions = () => {
+  const versions: { [key: string]: string } = {};
+  Object.entries(themes).forEach((theme) => {
+    const pkgn = theme[1].packageName;
+    const vers = execSync(`git tag --sort=taggerdate | grep "${pkgn}" | tail -1`).toString().trim();
+    versions[theme[0]] = vers.replace(/^@cmsgov\/.*@(.*)$/, '$1');
+  });
+  return versions;
+};
+const versions = getLatestVersions();
+
+export const getLatestCoreTag = () => {
+  return execSync(`git tag --sort=taggerdate | grep "${themes.core.packageName}" | tail -1`)
+    .toString()
+    .trim();
+};
+
+/**
+ * Get list of PR's associated with the current milestone formatted
+ * for our use as a PRDetails object.
+ */
+const getPRs = () => {
+  const prData = JSON.parse(
+    execSync(
+      `gh pr list --search "milestone:${cm.title}" --state merged --json title,url,labels,number,author,mergeCommit`
+    ).toString()
+  );
+  const prs: PRDetails[] = prData.map((pr: any) => {
+    return {
+      author: pr.author.login,
+      ghpr: pr.number,
+      hash: pr.mergeCommit.oid,
+      labels: pr.labels.map((label: any) => label.name),
+      ticket: pr.title.replace(/\[(.+)\](.*)/, '$1'),
+      title: pr.title.replace(/\[(.+)\]\s+(.*)/, '$2'),
+    };
+  });
+  return prs;
+};
+
+/**
+ * Organizes Note(s) by the category/section they should belong in.
+ * In order to keep this code as simple as possible, this just breaks up
+ * pr's by their impacts. Right now if an item impacts multiple items, it
+ * will be presented in each of those sections. An item will also be placed
+ * in multiple type categories if it belongs to multiple.
+ */
+const organizeNotes = (data: PRDetails[]) => {
+  const notes: any = [];
```

Comment:
It looks like the notes type is always the same and could be defined. Might help in the next function that consumes it. Looks like `[i, prType[0], pr.title, pr.ghpr]` is just a tuple of `[number, string, string, string]`

```ts
type PRNote = [number, string, string, string];
...
  const notes: PRNote = [];
```

Reply:
Making this variable a const requires you initialize it with data of its type, as `any[]` that type can be determined by the loop.  I'm not sure if there is a better way to do this.

Reply:
So `const notes: PRNote = [];` results in an error? If that were to be combined with the change below, would it result in an error? Just troubleshooting

```ts
notes.push([i, prType[0], pr.title, pr.ghpr] as any);
```

Reply:
Yeah `const declarations must be initialized`. Can't get around it.